string2.py

```diff
@@ -52,8 +60,25 @@ def not_bad(s):
 
 
 def front_back(a, b):
-    # your code here
-    return
+    alen = len(a)
+    blen = len(b)
+
+    if alen % 2 == 0:
+        aindex = alen // 2
+    else:
+        aindex = (alen //2) + 1
+    if blen % 2 == 0:
+        bindex = blen // 2
+    else:
+        bindex = (blen //2) + 1
+
+    astart = a[0:aindex]
+    aend = a[aindex:]
+
+    bstart = b[0:bindex]
+    bend = b[bindex:]
+
+    return astart + bstart + aend + bend
```

Comment:
Again, I'll note you should be using snake_case, but other than that, what I love about your solution here is how readable it is. I see your choices exactly and how the variables get put together in the end. 

You _could_ make the same choices with fewer lines of code. If you declared the "default" a_index and b_index on line 65, and then changed them only if they were odd, you'd be able to eliminate two else statements. 

Good work, Leanne!

Reply:
Thank you!!